common/ethereum/applications/chronos/chronos.go

```
@@ -0,0 +1,341 @@
+        "name": "to",
+        "type": "address"
+      }
+    ],
+    "name": "Swap",
+    "type": "event"
+  },
+  {
+    "inputs": [],
+    "name": "isStable",
+    "outputs": [
+      {
+        "internalType": "bool",
+        "name": "",
+        "type": "bool"
+      }
+    ],
+    "stateMutability": "view",
+    "type": "function"
+  },
+  {
+    "inputs": [],
+    "name": "token0",
+    "outputs": [
+      {
+        "internalType": "address",
+        "name": "",
+        "type": "address"
+      }
+    ],
+    "stateMutability": "view",
+    "type": "function"
+  },
+  {
+    "inputs": [],
+    "name": "token1",
+    "outputs": [
+      {
+        "internalType": "address",
+        "name": "",
+        "type": "address"
+      }
+    ],
+    "stateMutability": "view",
+    "type": "function"
+  }
+]`
+
+const chronosPairFactoryAbiString = `[
+  {
+  "inputs": [
+    {
+    "internalType": "bool",
+    "name": "_stable",
+    "type": "bool"
+    }
+  ],
+  "name": "getFee",
+  "outputs": [
+    {
+    "internalType": "uint256",
+    "name": "",
+    "type": "uint256"
+    }
+  ],
+  "stateMutability": "view",
+  "type": "function"
+  }
+]`
+
+var (
+	// chronosPairAbi set by init.go to generate the ABI code
+	chronosPairAbi ethAbi.ABI
+	// chronosPairFactoryAbi set by init.go to generathe the ABI code for the factory
+    chronosPairFactoryAbi ethAbi.ABI
+)
+
+// GetChronosFees returns Chronos' fee based on fees given by Chronos' pair factory
+// If the token swapped from was the fluid token, get the exact amount,
+// otherwise approximate the cost based on the received amount of the fluid token
+func GetChronosFees(transfer worker.EthereumApplicationTransfer, client *ethclient.Client, fluidTokenContract ethCommon.Address, tokenDecimals int) (*big.Rat, error) {
+	// decode the amount of each token in the log
+	// doesn't contain addresses, as they're indexed
+	if len(transfer.Log.Topics) < 1 {
+		return nil, fmt.Errorf("No log topics passed!")
```

Better to return the errors beginning with lowercase as per Go style